src/Settings.cpp

```diff
@@ -155,3 +159,24 @@ bool Settings::startInTray() const
 
     return enable;
 }
+
+void Settings::setAutoStart()
+{
+    std::string path{};
+    if(!Settings::instance().findDesktopFile(path))
+        return;
+
+    if(Settings::instance().getStartInTray())
+        std::system(("cp " + path + " ~/.config/autostart/").c_str());
+    else
+        std::system("rm ~/.config/autostart/whatsapp-for-linux.desktop");
+}
+
+bool Settings::findDesktopFile(std::string& outPath)
```

Comment:
As outPath is an out variable you should also return the string. For that, you can use a `std::pair<bool, std::string>`